### DISCUSSION TITLE
Add the ability to key menu items to avoid duplicate active states

### DISCUSSION THREAD

**Discussion by piljac1** [laravel/nova-issues#5110] [Ideas & Feature Requests]
We were recently asked to add a menu item that points to a resource index (which is also available in the menu) but with specific filters already applied. However, since the path is the same, both menu items (the resource and the custom menu item) have the `active` style.

It would be great if you could specify a menu key that would be sent as a query param that would later get compared in the active check. By default, menu items wouldn't have a key (so they would rely on the current `active` logic), but if a menu key is present, default menu items would automatically return `false` in the `active` method and only the menu item matching this key would return `true`.

Of course, slight changes would be needed in the `active` method to make it work, but the suggested implementation wouldn't be a breaking change, and therefore, would be possible to add in a minor release.

Let me know what you guys think,

Thanks!

**Comment by piljac1** [comment]
Bump

**Reply by steveHimself1397** [reply]
I second this.
I have two menu items /<some-resource> and /<some-resource>/export and when I am on the /export route I get both menu items highlighted.
So my guess is also, that the active check does not work correctly but I haven't found it yet in the source.
Any help would be appreciated
